diff --git a/core/modules/rest/tests/src/Functional/EntityResource/ConfigEntityResourceTestBase.php b/core/modules/rest/tests/src/Functional/EntityResource/ConfigEntityResourceTestBase.php
index fe3f858e794..75c37fa3dea 100644
--- a/core/modules/rest/tests/src/Functional/EntityResource/ConfigEntityResourceTestBase.php
+++ b/core/modules/rest/tests/src/Functional/EntityResource/ConfigEntityResourceTestBase.php
@@ -19,7 +19,7 @@ abstract class ConfigEntityResourceTestBase extends EntityResourceTestBase {
   /**
    * {@inheritdoc}
    */
-  public function setUp() {
+  protected function setUp() {
     if (in_array($this->getName(), static::SKIP_METHODS, TRUE)) {
       // Skip before installing Drupal to prevent unnecessary use of resources.
       $this->markTestSkipped("Not yet supported for config entities.");
diff --git a/core/modules/rest/tests/src/Functional/EntityResource/EntityResourceTestBase.php b/core/modules/rest/tests/src/Functional/EntityResource/EntityResourceTestBase.php
index bea7a29faeb..477f431dbcc 100644
--- a/core/modules/rest/tests/src/Functional/EntityResource/EntityResourceTestBase.php
+++ b/core/modules/rest/tests/src/Functional/EntityResource/EntityResourceTestBase.php
@@ -179,7 +179,7 @@ protected function provisionEntityResource($single_format = FALSE) {
   /**
    * {@inheritdoc}
    */
-  public function setUp() {
+  protected function setUp() {
     parent::setUp();
 
     // Calculate REST Resource config entity ID.
diff --git a/core/modules/rest/tests/src/Functional/ResourceTestBase.php b/core/modules/rest/tests/src/Functional/ResourceTestBase.php
index 575bf976dcb..a4569b31605 100644
--- a/core/modules/rest/tests/src/Functional/ResourceTestBase.php
+++ b/core/modules/rest/tests/src/Functional/ResourceTestBase.php
@@ -98,7 +98,7 @@ abstract class ResourceTestBase extends BrowserTestBase {
   /**
    * {@inheritdoc}
    */
-  public function setUp() {
+  protected function setUp() {
     parent::setUp();
 
     $this->serializer = $this->container->get('serializer');
